src/NuGet.Core/NuGet.Protocol/Providers/OwnerDetailsUriResourceV3Provider.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Resources;
+
+namespace NuGet.Protocol.Providers
+{
+    /// <summary>NuGet.Protocol resource provider for <see cref="OwnerDetailsUriTemplateResourceV3"/> in V3 HTTP feeds.</summary>
+    /// <remarks>When successful, returns an instance of <see cref="OwnerDetailsUriTemplateResourceV3"/>.</remarks>
+    public class OwnerDetailsUriResourceV3Provider : ResourceProvider
+    {
+        public OwnerDetailsUriResourceV3Provider()
+            : base(typeof(OwnerDetailsUriTemplateResourceV3),
+                  nameof(OwnerDetailsUriTemplateResourceV3),
+                  NuGetResourceProviderPositions.Last)
+        {
+        }
+
+        /// <inheritdoc cref="ResourceProvider.TryCreate(SourceRepository, CancellationToken)"/>
+        public override async Task<Tuple<bool, INuGetResource?>> TryCreate(SourceRepository source, CancellationToken token)
+        {
+            OwnerDetailsUriTemplateResourceV3? resource = null;
+            ServiceIndexResourceV3? serviceIndex = await source.GetResourceAsync<ServiceIndexResourceV3>(token);
+            if (serviceIndex != null)
+            {
+                Uri? uriTemplate = serviceIndex.GetServiceEntryUri(ServiceTypes.OwnerDetailsUriTemplate);
+
+                if (uriTemplate != null)
+                {
+                    resource = OwnerDetailsUriTemplateResourceV3.CreateOrNull(uriTemplate);
```

Comment:
I am also wondering if we should invoke `uriTemplate.IsWellFormedOriginalString()` as per [security considerations](https://learn.microsoft.com/dotnet/fundamentals/runtime-libraries/system-uri#security-considerations) in the docs.

Reply:
I ran a test and I'm still seeing that the template will never be considered well formed, and that the generated URI, even with `...` is indeed considered well formed. Can you provide a test case here that shows what this will catch?

> URL: https://nuget.test/profiles/{owner}?_src=template
> Valid?True
> IsWellFormed?False
> 
> URL: https://nuget.test/profiles/donnie?_src=template
> Valid?True
> IsWellFormed?True
> 
> URL: https://nuget.test/profiles/../?_src=template
> Valid?True
> IsWellFormed?True

Reply:
Talked offline - resolving